packages/entities/entities-gateway-services/src/components/GatewayServiceForm.vue

```diff
@@ -587,34 +672,248 @@ const changeCheckedGroup = () => {
   form.fields.tls_verify_value = formFieldsOriginal.tls_verify_value
 }
 
-const validateUrl = (): void => {
-  if (form.fields.url && checkedGroup.value === 'url') {
+const changeCheckedGroup = () => {
+  isCollapsed.value = true
+  resetFormFieldErrors()
+  form.errorMessages = []
+  initFieldDefaultValues()
+}
+
+const handleTrySampleApi = (): void => {
+  // Increment demo try counter
+  demoTryTimes.value += 1
+
+  // Determine which demo service to use based on even/odd counter
+  const isWeatherDemo = demoTryTimes.value % 2 === 0
+  const serviceRecord = isWeatherDemo ? { ...demoWeatherService } : { ...demoStockService }
+
+  // Reset form fields before populating
+  initFieldDefaultValues()
+
+  form.fields.name = serviceRecord.name
+
+  switch (checkedGroup.value) {
+    case 'url':
+      // Construct the full URL
+      form.fields.url = `${serviceRecord.protocol}://${serviceRecord.host}${serviceRecord.path}`
+      break
+
+    case 'protocol':
+      // Populate individual protocol-related fields
+      form.fields.host = serviceRecord.host
+      form.fields.path = serviceRecord.path
+      form.fields.protocol = serviceRecord.protocol
+      form.fields.port = serviceRecord.port
+      break
+
+    default:
+      console.warn('Invalid input group selected')
+  }
+}
+
+const validateHost = (host: string): string => {
+  // check if host is empty
+  if (!host || host.trim() === '') return 'host cannot be empty'
+
+  // check if a valid host (domain or ip address)
+  const hostnameRegex = /^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9-]*[A-Za-z0-9])$|^\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}$/
+  if (!hostnameRegex.test(host)) return 'invalid host'
+  return ''
+}
+
+const validateProtocol = (protocol: string) => {
```

Comment:
suggestion: move all of these validator functions into a composable. Add unit tests.